contracts/contracts/strategies/balancer/BalancerComposablePoolStrategy.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title OETH Balancer ComposableStablePool Strategy
+ * @author Origin Protocol Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { BalancerMetaPoolStrategy } from "./BalancerMetaPoolStrategy.sol";
+import { IERC20 } from "../../utils/InitializableAbstractStrategy.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+
+contract BalancerComposablePoolStrategy is BalancerMetaPoolStrategy {
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        BaseBalancerConfig memory _balancerConfig,
+        BaseMetaPoolConfig memory _metapoolConfig,
+        address _auraRewardPoolAddress
+    )
+        BalancerMetaPoolStrategy(
+            _stratConfig,
+            _balancerConfig,
+            _metapoolConfig,
+            _auraRewardPoolAddress
+        )
+    {}
+
+    function _assetConfigVerification(address[] calldata _assets)
+        internal
+        view
+        override
+    {
+        IERC20[] memory poolAssets = _getPoolAssets();
+
+        require(
+            // aside from BPT token all assets must be supported
+            poolAssets.length - 1 == _assets.length,
```

Comment:
nit: Minor gas savings by storing `_assets.length` in a local variable. Same with `_amounts.length` in the following functions as well